src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -241,6 +241,48 @@ namespace OpenMS
     spectrum.getPrecursors().push_back(prec);
   }
 
+  MSSpectrum TheoreticalSpectrumGenerator::generateSpectrum(const Precursor::ActivationMethod& fm, const AASequence& seq, int precursor_charge)
+  {
+    if (precursor_charge == 0)
+    {
+      OPENMS_LOG_WARN << "Precursor charge can't be 0. Using 1 instead." << endl;
```

Comment:
hmm not sure if this is the best choice? others?
the most prominent precursor charge for peptides is usually 2